src/catalog.h

```diff
@@ -28,8 +28,17 @@ enum CatalogTable
 	_MAX_CATALOG_TABLES,
 };
 
+typedef enum InternalFunctions
```

Comment:
Nit: I prefer that enums are in singular form, i.e., "InternalFunction". Makes more sense when you pass that type as, e.g., a parameter since every instance of the enum type represents only one function.